python/Makefile.am

```diff
@@ -166,23 +168,40 @@ endif
 
 INIT_PY = __init__.py
 MPB_PY_PATH = $(builddir)/mpb.py
-else
+
+else # not maintainer mode
 
 INIT_PY = $(srcdir)/__init__.py
 MPB_PY_PATH = $(srcdir)/mpb.py
 
 endif # MAINTAINER_MODE
 
+######################################################################
+# specification of python source files to be byte-compiled at installation
+######################################################################
 HL_IFACE =                  \
     $(srcdir)/geom.py       \
     $(srcdir)/simulation.py \
     $(srcdir)/source.py     \
     $(srcdir)/materials.py
 
-PY_PKG_FILES =      \
-    $(INIT_PY)      \
-    $(HL_IFACE)     \
-    .libs/_meep.so
+pkgpython_PYTHON = __init__.py $(HL_IFACE)
+
+adjointdir = $(pkgpythondir)/adjoint
+adjoint_PYTHON  = $(srcdir)/adjoint/__init__.py			\
+                  $(srcdir)/adjoint/Basis.py  			\
+                  $(srcdir)/adjoint/Objective.py        	\
+                  $(srcdir)/adjoint/OptimizationProblem.py	\
+                  $(srcdir)/adjoint/ParallelDesignTester.py	\
+                  $(srcdir)/adjoint/Visualization.py
+
+######################################################################
+# finally, specification of what gets installed in the meep python
+# module directory of the python site-packages installation
+# Q: Why is this not redundant since e.g. the HL_IFACE files should
+#    already be installed by virtue of being in pkgpython_PYTHON
```

Comment:
This section assembles the `meep` package hierarchy in the build directory so that the normal 
```bash
make
make check
make install
```
workflow tests the built package before installing anything. If there were any adjoint solver tests, `make check` would currently be unable to import the `meep.adjoint` module.